mmv1/templates/terraform/examples/firebase_app_hosting_backend_basic.tf.tmpl

```diff
@@ -0,0 +1,76 @@
+# Service account setup only required once per project.
+resource "google_service_account" "service_account" {
+  provider = google-beta
+
+  project = "{{index $.TestEnvVars "project_id"}}"
+
+  # Must be firebase-app-hosting-compute
+  account_id                   = "firebase-app-hosting-compute"
+  display_name                 = "Firebase App Hosting compute service account"
+
+  # Do not throw if already exists
+  create_ignore_already_exists = true
+}
+
+resource "google_project_iam_member" "app_hosting_sa_developerconnect" {
+  provider = google-beta
+
+  project = "{{index $.TestEnvVars "project_id"}}"
+
+  # For reading connected Github repos
+  role   = "roles/developerconnect.readTokenAccessor"
+  member = google_service_account.service_account.member
+}
+
+resource "google_project_iam_member" "app_hosting_sa_adminsdk" {
+  provider = google-beta
+
+  project = "{{index $.TestEnvVars "project_id"}}"
+
+  # For Firebase Admin SDK
+  role   = "roles/firebase.sdkAdminServiceAgent"
+  member = google_service_account.service_account.member
+}
+
+resource "google_project_iam_member" "app_hosting_sa_runner" {
+  provider = google-beta
+
+  project = "{{index $.TestEnvVars "project_id"}}"
+
+  # For App Hosting
+  role   = "roles/firebaseapphosting.computeRunner"
+  member = google_service_account.service_account.member
+}
+
+# Service enablement only required once per project.
+resource "google_project_service" "fah" {
+  provider = google-beta
+
+  project = "{{index $.TestEnvVars "project_id"}}"
+  service = "firebaseapphosting.googleapis.com"
+
+  disable_on_destroy = false
+}
+
+resource "google_firebase_app_hosting_backend" "example" {
```

Comment:
This being the only test I see 2 main issues:
* We never check if optional fields can be omitted. This would normally be done by having one of the test configs be minimal (only required fields), and another with all possible fields specified.
* We never check if mutable fields can be changed (like app_id, service_account, etc.). Unless there is some limitation, we should use an [update test](https://googlecloudplatform.github.io/magic-modules/test/test/#add-an-update-test) for them.

Reply:
Right, added.